File: app/src/test/java/com/mecong/tenderalarm/AlarmUtilsTest.java
package com.mecong.tenderalarm;

import com.mecong.tenderalarm.model.AlarmEntity;

import org.junit.jupiter.api.Test;

import java.util.Calendar;

import static org.junit.jupiter.api.Assertions.assertEquals;

class AlarmUtilsTest {

    @Test
    void getNextAlarmDate() {

        AlarmEntity entity = new AlarmEntity();
        entity.setHour(17);
        entity.setMinute(54);
        entity.setDays(0b00000000);

        Calendar expectedCalendar = Calendar.getInstance();
        expectedCalendar.set(Calendar.HOUR_OF_DAY, 17);
        expectedCalendar.set(Calendar.MINUTE, 54);

        entity.updateNextAlarmDate(true);
        Calendar actualCalendar = Calendar.getInstance();
        actualCalendar.setTimeInMillis(entity.getNextTime());

        assertEquals(expectedCalendar.get(Calendar.HOUR_OF_DAY), actualCalendar.get(Calendar.HOUR_OF_DAY));
        assertEquals(expectedCalendar.get(Calendar.MINUTE), actualCalendar.get(Calendar.MINUTE));
    }
}

File: app/src/test/java/com/mecong/tenderalarm/AlarmUtilsTest.kt
package com.mecong.tenderalarm

import com.mecong.tenderalarm.model.AlarmEntity
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.*

internal class AlarmUtilsTest {
  @Test
  fun test_getNextAlarmDate() {
    val entity = AlarmEntity()
    entity.hour = 17
    entity.minute = 54
    entity.days = 0
    val expectedCalendar = Calendar.getInstance()
    expectedCalendar[Calendar.HOUR_OF_DAY] = 17
    expectedCalendar[Calendar.MINUTE] = 54
    entity.updateNextAlarmDate(true)
    val actualCalendar = Calendar.getInstance()
    actualCalendar.timeInMillis = entity.nextTime
    Assertions.assertEquals(expectedCalendar[Calendar.HOUR_OF_DAY], actualCalendar[Calendar.HOUR_OF_DAY])
    Assertions.assertEquals(expectedCalendar[Calendar.MINUTE], actualCalendar[Calendar.MINUTE])
  }
}